shared/serdes/index.js

```diff
@@ -0,0 +1,126 @@
+const raw = Symbol('raw')
+export const serdesTagSymbol = Symbol('tag')
+export const serdesSerializeSymbol = Symbol('serialize')
+export const serdesDeserializeSymbol = Symbol('deserialize')
+
+// This file is at attempt at addressing the lack of support for
+// custom objects in `structuredClone`. See <https://github.com/whatwg/html/issues/7428>.
+// We need this so that certain custom objects can be shared using a
+// `MessagePort`. To use this functionality, the `serializer` function must be
+// called on the _sending_ side and the `deserializer` function must be called
+// on the _receiving_ side.
+// Note that it's paramount that the _receiving_ side call `deserializer.register`
+// with all of the possible types that are supported.
+// For how to implement `serialize` and `deserialize` support for custom types,
+// see the example below for a class `X`.
+
+// Internal function to mark the result of 'serializer' as internal, so that it
+// doesn't get accidentally reprocessed.
+const rawResult = (obj: Object) => {
+  Object.defineProperty(obj, raw, { value: true })
+  return obj
```

Comment:
Could you:

- Add a comment explaining why `Object.defineProperty` is used instead of `obj[raw] = true`?
- I think this can be shortened to a 1-liner if you put the `return` in front of the `Object` on line 20

Reply:
> this can be shortened to a 1-liner if

I don't think that contributes to readability, because it's not self-evident that `Object.defineProperty(a, b)` returns `a`.